### ISSUE TITLE
Firefox and Vivaldi Browsers use Nemo rather than Dolphin to open folder of a downloaded file

### ISSUE THREAD

**Issue by TraceyC77** [getsolus/packages#1149]
### Summary

On Plasma, with both Dolphin and Nemo installed, browsers use Nemo for opening the folder of a download.

### Steps to reproduce

1. On Plasma, install Nemo from Software Center
2. Verify that Dolphin is still the default app for "File manager" in Default Applications
3. Using either Firefox or Vivaldi, download a file. When it's done, try to open the folder it was downloaded to (In Firefox this is via Downloads - click on the Folder icon (Show in Folder))

### Expected result

The folder is opened in Dolphin

### Actual result

The folder is opened in Nemo

### Environment

- [X] Is system up to date?

### Repo

Unstable

### Desktop Environment

Plasma

### System details

Operating System: Solus 4.5
KDE Plasma Version: 5.27.10
KDE Frameworks Version: 5.112.0
Qt Version: 5.15.11
Kernel Version: 6.6.8-268.current (64-bit)
Graphics Platform: X11
Processors: 16 × 11th Gen Intel® Core™ i7-11800H @ 2.30GHz
Memory: 62.5 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3060 Laptop GPU/PCIe/SSE2
Manufacturer: Dell Inc.
Product Name: XPS 17 9710

### Other comments

I just noticed this today, and I have had Nemo installed for a while (to test another Nemo issue).
I found [an issue in the Mozilla tracker](https://bugzilla.mozilla.org/show_bug.cgi?id=266600) that says:

> If using Dolphin as the default filemanager, and it responds to the org.freedesktop.FileManager1 dbus interface (this is the default setup in KDE AFAIK), Dolphin gets used appropriately and everything is fine.
 
    If your default filemanager actively listens/responds to the org.freedesktop.FileManager1 dbus interface, it works as expected
    If your default filemanager supports org.freedesktop.FileManager1 but doesn't actively listen/respond to it, it depends
    If your default filemanager doesn't support org.freedesktop.FileManager1 but other installed filemanager(s) do, it's still broken
    If your default filemanager doesn't support org.freedesktop.FileManager1 and no other installed filemanager(s) do (or you overrule the service so nothings responds to it), it works as expected

**Comment by Justinzobel** [comment]
I can't replicate this on Plasma 6. Does it still happen for you on 5? If it does, can you try (if you have a test device/VM) with Plasma 6?

**Comment by TraceyC77** [comment]
Yes, I was able to reproduce this just now with Firefox and Vivaldi

**Comment by Justinzobel** [comment]
Can confirm in a Plasma 5 VM.

**Comment by malfisya** [comment]
Maybe related https://github.com/getsolus/packages/issues/1470

**Comment by TraceyC77** [comment]
Tested on Plasma 6.0.1
I can reproduce the bug. 

We probably need to add a default `mimeapps.list` file to the Plasma branding, similar to in #1470

**Comment by TraceyC77** [comment]
This is now resolved.
@ReillyBrogan , please link this to the correct PR  / commit / wherever the fix is and close.
I would do this but I don't see a related PR?

**Comment by TraceyC77** [comment]
Reilly told me this was an upstream fix

**Comment by TraceyC77** [comment]
Re-opening since yesterday, Firefox opened folders with Dolphin. Today it's nemo.

**Comment by ReillyBrogan** [comment]
Try again

**Comment by TraceyC77** [comment]
I just downloaded a file with Firefox and opened the folder. It opened with Nemo.

**Comment by ermo** [comment]
If this has been fixed, move back to 4.6 and close it there.

If not, leave it in 4.7 for now.

**Comment by TraceyC77** [comment]
This is still an issue

**Comment by ReillyBrogan** [comment]
If you create a new user does the issue still happen?

**Comment by ReillyBrogan** [comment]
What is the output of `xdg-mime query default inode/directory`

**Comment by TraceyC77** [comment]
Interestingly enough, on my Flex with plasma built from git-master I see the same behavior

On that machine and my main (XPS) I find the behavior differs depending on whether the Plasma Integration addon is enabled.

- With a Firefox profile that has no addons, opening the folder of a download opens it in Nemo
- On the same machine, with a Firefox profile that has Plasma Integration enabled, the folder open in Dolphin
This behavior is the same on both machines

**Comment by ReillyBrogan** [comment]
I think this is a Firefox-specific bug. As far as I can tell it's bypassing the default mimetypes handling logic.

**Comment by ReillyBrogan** [comment]
The KDE integration patches that SUSE maintains don't help unfortunately.

**Comment by TraceyC77** [comment]
> I think this is a Firefox-specific bug. As far as I can tell it's bypassing the default mimetypes handling logic.

The same problem happens in Vivaldi as well

**Comment by TraceyC77** [comment]
I found an upstream bug in the KDE tracker for this. It affects other things in addition to opening download folders from the browsr.

[Bug 397953](https://bugs.kde.org/show_bug.cgi?id=397953)
KIO::highlightInFileManager does not respect default file manager setting when launched via FileManager1 DBus activation

I'm going to close this out again, since this needs to be solved by the KDE devs

**Comment by TraceyC77** [comment]
I also found this in the Mozilla bug tracker. Including for posterity.
https://bugzilla.mozilla.org/show_bug.cgi?id=1285711
